app/main.py

```diff
@@ -1,2 +1,242 @@
+from __future__ import annotations
+
+import copy
+import dataclasses
+from typing import Any, Callable, Iterable, Hashable
+
+
+@dataclasses.dataclass
+class Node:
```

Comment:
I would define this class inside of a Dictionary since it is a part of its implementation and cannot be used outside it

Reply:
This hasn't been done.